### PR TITLE
hv: fix coding style violations in acpi.c

### PR DESCRIPTION
**Pull request by yonghuah**
1. to avoid data type cast when using 'global_rsdp'
 2. other minor fix

Tracked-On: #861
Signed-off-by: Yonghua Huang <yonghua.huang@intel.com>
Acked-by: Anthony Xu <anthony.xu@intel.com>

### CODE REVIEW

File: hypervisor/boot/acpi.c [hunk 1]
```diff
@@ -100,7 +100,7 @@ struct acpi_madt_local_apic {
 	uint32_t                       lapic_flags;
 };
 
-static void *global_rsdp;
+static struct acpi_table_rsdp *global_rsdp;
```
Comment (dongyaozu) [inline hypervisor/boot/acpi.c]: Why this is called as "global" ?  It is confusing for a static variable to be named as global.

Comment (yonghuah) [reply hypervisor/boot/acpi.c]: @dongyaozu,  thanks, have updated, pls review.

**Comment by ClaudZhang1995** [comment]
Ok to verify

**Comment by ClaudZhang1995** [comment]
Ok to verify

**Comment by wenlingz** [comment]
OK to verify

**Comment by HuiHuangShi** [comment]
2 new potential violations to the coding guideline detected. Please help check. Thanks!

1.filename:/hypervisor/boot/acpi.c  function:parse_madt  offset:259:
       reason:Casting operation to a pointer. : (void* to struct acpi_table_madt*): ( struct acpi_table_madt * ) get_acpi_tbl ( "APIC" )

2.filename:/hypervisor/boot/acpi.c  function:parse_madt  offset:259:
       reason:Casting operation on a pointer. : (void* to struct acpi_table_madt*): ( struct acpi_table_madt * ) get_acpi_tbl ( "APIC" )

**Comment by HuiHuangShi** [comment]
> 2 new potential violations to the coding guideline detected. Please help check. Thanks!
> 
> 1.filename:/hypervisor/boot/acpi.c function:parse_madt offset:259:
> reason:Casting operation to a pointer. : (void* to struct acpi_table_madt*): ( struct acpi_table_madt * ) get_acpi_tbl ( "APIC" )
> 
> 2.filename:/hypervisor/boot/acpi.c function:parse_madt offset:259:
> reason:Casting operation on a pointer. : (void* to struct acpi_table_madt*): ( struct acpi_table_madt * ) get_acpi_tbl ( "APIC" )

1 and 2 should be deviated, will fill in the justification later.

**Comment by ClaudZhang1995** [comment]
Ok to verify

**Comment by ClaudZhang1995** [comment]
Ok to verify

**Pull request merged**